substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -252,16 +234,18 @@ fn check_event_type(
 			// Check bound contains IsType and From
 
 			let has_is_type_bound = type_.bounds.iter().any(|s| {
-				syn::parse2::<IsTypeBoundEventParse>(s.to_token_stream())
-					.map_or(false, |b| b.0 == *frame_system)
+				syn::parse2::<IsTypeBoundEventParse>(s.to_token_stream()).map_or(false, |b| {
+					let mut expected_system_config = frame_system.clone();
+					expected_system_config
+						.segments
+						.push(syn::PathSegment::from(syn::Ident::new("Config", b.0.span())));
+					b.0 == expected_system_config
```

Comment:
Please double-check this condition for correctness, and if no tests exists for it, please add one :)